ble/src/u_ble_sps_intmod.c

```diff
@@ -1253,13 +1258,13 @@ int32_t uBleSpsSend(uDeviceHandle_t devHandle, int32_t channel, const char *pDat
                 // again later if we are out of credits.
                 (void)uPortSemaphoreTryTake(pSpsConn->txCreditsSemaphore, 0);
                 if (pSpsConn->txCredits == 0) {
-                    int32_t timeoutLeft = (int32_t)timeout - (int32_t)(time - startTime);
+                    int32_t timeoutLeft = timeout - (time - startTimeMs);
```

Comment:
Alternative implementation I think it does the same thing:

uint32_t elapsedMs = uTimeoutElapsdMs(startTime())
uint32_t timoutLeftMs = (elapsedMs >= timoutMs) ? 0 : timoutMs - elapsedMs;

Reply:
and remove in line 1285-1287

if (bytesLeftToSend > 0) {
    time = uPortGetTickTimeMs();
}